packages/driver/types/internal-types-lite.d.ts

```diff
@@ -33,6 +33,7 @@ declare namespace Cypress {
     isStable: IStability['isStable']
     fail: (err: Error, options:{ async?: boolean }) => Error
     getRemoteLocation: ILocation['getRemoteLocation']
+    subjectChain: (chainerId?: string) => any[]
```

Comment:
I assume we can't be more specific on the return types because of `cy.wrap()`?

Reply:
Fixed the type here to properly use `SubjectChain`. Also added proper types for `subjects` and `subjectLinks` to state.ts, as I was poking around trying to understand our types.

I don't understand internal-types-lite.d.ts; some files seem perfectly happy to reference the $Cy class definition, while other files need the methods to be defined here as well. I added the types here so that `check-ts` would pass, but I do not understand why I did that. :woman_shrugging: